tests/actions/globalPropertyEncounterType.test.js

```diff
@@ -0,0 +1,109 @@
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import moxios from 'moxios';
+import {
+    globalPropertyEncounterTypeSuccess,
+    globalPropertyEncounterTypeLoading,
+    globalPropertyEncounterTypeFailure,
+    getGlobalPropertyEncounterType,
+} from '../../app/js/actions/globalPropertyEncounterType';
+import {
+    GLOBAL_PROPERTY_ENCOUNTER_TYPE_SUCCESS,
+    GLOBAL_PROPERTY_ENCOUNTER_TYPE_FAILURE,
+    GLOBAL_PROPERTY_ENCOUNTER_TYPE_LOADING,
+} from '../../app/js/actions/actionTypes';
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+describe ('Get the encounterType configuration global property actions', () => {
+    it('should create an action to store fetched configuration', () => {
+        const configuration = 'order entry';
+        const expectedAction = {
+            type: GLOBAL_PROPERTY_ENCOUNTER_TYPE_SUCCESS,
+            configuration
+        };
+        expect(globalPropertyEncounterTypeSuccess(configuration)).toEqual(expectedAction);
+    });
+
+    it('should create an action to log failure', () => {
+        const error = 'No network connection';
+        const expectedAction = {
+            type: GLOBAL_PROPERTY_ENCOUNTER_TYPE_FAILURE,
+            error
+        };
+        expect(globalPropertyEncounterTypeFailure(error)).toEqual(expectedAction);
+    });
+
+    it('should create action for loading event', () => {
+        const status = true;
+        const expectedAction = {
+            type: GLOBAL_PROPERTY_ENCOUNTER_TYPE_LOADING,
+            status
+        };
+        expect(globalPropertyEncounterTypeLoading(status)).toEqual(expectedAction);
+    });
+
+    describe ('Get global property encounterType API call', () => {
```

Comment:
Why nest your Test Suites?

Reply:
The next tests had pre-conditions that were common to them, therefore I nested them in a suit so as to set and unset the preconditions at once.